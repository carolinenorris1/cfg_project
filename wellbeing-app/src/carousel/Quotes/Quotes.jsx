import React, { useState, useEffect } from "react";
import fetchQuote from "./QuotesAPI";

function Quotes() {
  const [quote, setQuote] = useState("");
  const [author, setAuthor] = useState("");

  useEffect(() => {
    fetchQuoteData();
  }, []);

  const fetchQuoteData = async () => {
    const result = await fetchQuote();
    if (result) {
      setQuote(result.content);
      setAuthor(result.originator.name);
    }
  };

  return (
    // Display the fetched quote at the top
    <div>
      <p>Quote for today:</p>
      <h2 id="quote">{quote}</h2>
      <p id="author">â€” {author}</p>
    </div>
  );
}

export default Quotes;
