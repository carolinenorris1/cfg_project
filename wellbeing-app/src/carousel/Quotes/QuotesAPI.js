const fetchQuote = async () => {
    const url = 'https://quotes15.p.rapidapi.com/quotes/random/?language_code=en';
    const options = {
      method: 'GET',
      headers: {
        'X-RapidAPI-Key': 'b5a8fac0e4msh68e0442b43ca2d6p137abdjsn64fea5244fc4',
        'X-RapidAPI-Host': 'quotes15.p.rapidapi.com'
      }
    };
  
    try {
      const response = await fetch(url, options);
      const result = await response.json();
      return result;
    } catch (error) {
      console.error('Error fetching data:', error);
      return null;
    }
  };
  
  export default fetchQuote;