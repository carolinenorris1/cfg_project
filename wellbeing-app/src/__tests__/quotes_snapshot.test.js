import React from "react";
import renderer from "react-test-renderer";

import Quotes from "../carousel/Quotes/Quotes";

it("renders correctly when user views the daily quote on the carousel", () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
