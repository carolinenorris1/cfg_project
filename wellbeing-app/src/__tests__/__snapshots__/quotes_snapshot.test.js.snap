// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when user views the daily quote on the carousel 1`] = `
<div>
  <p>
    Quote for today:
  </p>
  <h2
    id="quote"
  />
  <p
    id="author"
  >
    â€” 
  </p>
</div>
`;
